pkg/cloudinfo/alibaba/cloudinfo.go

```diff
@@ -121,50 +82,60 @@ func NewAliInfoer(ctx context.Context, cfg Config) (*AlibabaInfoer, error) {
 }
 
 // Initialize is not needed on Alibaba because price info is changing frequently
-func (e *AlibabaInfoer) Initialize(ctx context.Context) (map[string]map[string]cloudinfo.Price, error) {
+func (a *AlibabaInfoer) Initialize(ctx context.Context) (map[string]map[string]cloudinfo.Price, error) {
 	return nil, nil
 }
 
-func (e *AlibabaInfoer) getCurrentSpotPrices(ctx context.Context, region string, zones []string) (map[string]cloudinfo.SpotPriceInfo, error) {
+func (a *AlibabaInfoer) getCurrentSpotPrices(ctx context.Context, region string) (map[string]cloudinfo.SpotPriceInfo, error) {
 	log := logger.Extract(ctx)
 	log.Debug("start retrieving spot price data")
 	priceInfo := make(map[string]cloudinfo.SpotPriceInfo)
 
-	request := ecs.CreateDescribeSpotPriceHistoryRequest()
-	request.RegionId = region
-	request.NetworkType = "vpc"
-	request.OSType = "linux"
+	request := requests.NewCommonRequest()
```

Comment:
assembling reqests and other boilerplate could be extracted in a factory-like method or function